presenters/thread_utils.rb

```diff
@@ -17,6 +17,23 @@ def self.get_endorsed(threads)
     endorsed_threads
   end
 
+  def self.get_abuse_flagged_count(threads)
+    # returns a count of flagged threads for all comments within the thread
+    # only threads with at least one flag are returned
+    flagged_threads = {}
+    thread_ids = threads.collect {|t| t._id}
+    Comment.collection.aggregate([
+      {"$match" => {
+        "comment_thread_id" => {"$in" => thread_ids},
+        "abuse_flaggers"=> {"$ne"=> []},
+      }},
+      {"$group" => {"_id" => "$comment_thread_id", "flagged" => { "$sum" => 1}} }
+    ]).each do |res|
```

Comment:
Would love it if someone with knowledge of mongodb could examine this query and see if it's OK and if there are ways to optimise it. It is only run when this data is explicitly requested, but that's every time a moderator requests it.

Reply:
Thanks for the detail. A few questions to help understand it a little better before testing. 

1. Can you tell me How /when I'll hit this method? 
2. How many API URLs hit this method? 
3. Is see this is behind a condition `if count_flagged:` < is this condition currently in use in production? I mean can this API get this flag from the discussion UI today or it is only implemented for the new discussion UI?

Reply:
1. This will be hit under the following conditions: 
    - You are using the discussions API at `/api/discussion/v1/` AFAIK this is only used for mobile, the current UI uses a different API
    - You fetch one or more threads for the course using `http://localhost:18000/api/discussion/v1/threads`
    - You access the API as someone with moderator or above preveleges.
2. Just 1 
3. This is currently not used in produciton, however `cs_comments_service` APIs are not directly consumed. These are consumed by `edx-platform` which in turn will call these APIs. With the corresponding `edx-platform` change this condition will be true in production when using the above API which is only used by mobile clients and the new frontend.

Reply:
@xitij2000 I was thinking if we can debug the performance on production, in that case, do you have any ideas how we can do that?

Reply:
I'm not sure if this service is accessible in a way that it can be pinged, but if it is, you can run this query by doing the following:

`<cs_comments_host>/api/v1/threads?api_key={api_key}&course_id={test_course}&count_flagged=true`

This will call this API. If that's not possible, you fetch the following as a moderator or staff user:

`{LMS_HOST}/api/discussion/v1/threads/?course_id={test_course}`

Reply:
Plan:
1. @xitij2000 Will share Mongo query to run to read replica.
2. Will update the edx-platform PR to not send this flag automatically.

Reply:
@awaisdar001 
For (1) here is a similar mongo query:
```mongo
use cs_comments_service;
db.contents.aggregate([
    {
        $match: {
            course_id: "course-v1:edX+DemoX+Demo_Course", 
            abuse_flaggers: { $ne: [] }
        }
    }, {
        $group: {
            _id: "$comment_thread_id", 
            flagged: { $sum: 1 }
        } 
    }
]);
```

Note that I am asssuming the database name for the forums is cs_comments_service, because it is in the devstack. It may not be in this case.

Reply:
thanks, I am taking a look.